lightningd/jsonrpc.c

```diff
@@ -429,9 +429,8 @@ static void json_command_malformed(struct json_connection *jcon,
 				   const char *id,
 				   const char *error)
 {
-	return connection_result(jcon, id, NULL, error,
-				 JSONRPC2_INVALID_REQUEST,
-				 NULL);
+	return connection_result(jcon, id, NULL, tal_fmt(jcon, "\"%s\"", error),
```

Comment:
Would not this leak? Client applications might keep the `jcon` alive indefinitely. Of course, arguably an application should not trigger this code.

Arguably we should be using `json_add_string` somehow, which performs proper escaping, but I suppose this is okay.

Reply:
True.  We could make connection_result build the JSON response using primitives rather than manually...

But note that the other caller to connection_result() which has non-null error also does quoting manually.  We should do this inside connection_result()!